File: objectiveai-sdk-js/src/functions/profiles/computations/http.ts
import { ObjectiveAI, type RequestOptions } from "../../../client";
import { Stream } from "../../../stream";
import type { FunctionsProfilesComputationsRequestFunctionProfileComputationCreateParams } from "./request/functionProfileComputationCreateParams";
import type { FunctionsProfilesComputationsResponseUnaryFunctionProfileComputation } from "./response/unary/functionProfileComputation";
import type { FunctionsProfilesComputationsResponseStreamingFunctionProfileComputationChunk } from "./response/streaming/functionProfileComputationChunk";

export function functionsProfilesComputationsComputeProfile(
  client: ObjectiveAI,
  body: FunctionsProfilesComputationsRequestFunctionProfileComputationCreateParams & { stream: true },
  options?: RequestOptions,
): Promise<Stream<FunctionsProfilesComputationsResponseStreamingFunctionProfileComputationChunk>>;
export function functionsProfilesComputationsComputeProfile(
  client: ObjectiveAI,
  body: FunctionsProfilesComputationsRequestFunctionProfileComputationCreateParams & { stream?: false | null },
  options?: RequestOptions,
): Promise<FunctionsProfilesComputationsResponseUnaryFunctionProfileComputation>;
export function functionsProfilesComputationsComputeProfile(
  client: ObjectiveAI,
  body: FunctionsProfilesComputationsRequestFunctionProfileComputationCreateParams,
  options?: RequestOptions,
): Promise<
  | Stream<FunctionsProfilesComputationsResponseStreamingFunctionProfileComputationChunk>
  | FunctionsProfilesComputationsResponseUnaryFunctionProfileComputation
> {
  if (body.stream) {
    return client.post_streaming<FunctionsProfilesComputationsResponseStreamingFunctionProfileComputationChunk>(
      "functions/profiles/compute",
      body,
      options,
    );
  }
  return client.post_unary<FunctionsProfilesComputationsResponseUnaryFunctionProfileComputation>(
    "functions/profiles/compute",
    body,
    options,
  );
}

File: objectiveai-sdk-py/objectiveai_sdk/functions/profiles/computations/http.py
"""HTTP functions for profile computation endpoints."""

from __future__ import annotations

from typing import TYPE_CHECKING, Union

if TYPE_CHECKING:
    from objectiveai.client import ObjectiveAI
    from objectiveai.functions.profiles.computations.request import (
        FunctionProfileComputationCreateParams,
    )
    from objectiveai.functions.profiles.computations.response.streaming import (
        FunctionProfileComputationChunk,
    )
    from objectiveai.functions.profiles.computations.response.unary import (
        FunctionProfileComputation,
    )
    from objectiveai.stream import Stream


async def compute_profile(
    client: ObjectiveAI,
    params: FunctionProfileComputationCreateParams,
) -> Union[FunctionProfileComputation, Stream[FunctionProfileComputationChunk]]:
    """Compute a profile for a function.

    If ``params.stream`` is true, returns a streaming response.
    """
    if getattr(params, "stream", None):
        return await client.post_streaming("functions/profiles/compute", params)
    return await client.post_unary("functions/profiles/compute", params)
